clients/v1_repl/v1_repl.go

```diff
@@ -0,0 +1,921 @@
+			break
+		}
+	}
+	if bucketID == "" {
+		return "", fmt.Errorf("unable to find bucket ID for %q.%q", db, rp)
+	}
+	return bucketID, nil
+}
+
+type FormatType string
+
+var (
+	CsvFormat   FormatType = "csv"
+	JsonFormat  FormatType = "json"
+	TableFormat FormatType = "table"
+)
+
+func (c *Client) runAndShowQuery(query string) {
+	// ? Should this function have support to guide users trying to use deprecated InfluxQL queries
+	response, err := c.query(context.Background(), query)
+	if err != nil {
+		color.HiRed("Query failed.")
+		color.Red("%v", err)
+		return
+	}
+	displayMap := map[FormatType]func(string){
+		CsvFormat:   c.outputCsv,
+		JsonFormat:  c.outputJson,
+		TableFormat: c.outputTable,
+	}
+	displayFunc := displayMap[c.Format]
+	displayFunc(response)
+}
+
+// This function generates the prompt autocompletions
+func (c *Client) completer(d prompt.Document) []prompt.Suggest {
+	currentLineUpper := strings.ToUpper(d.CurrentLine())
+	var s []prompt.Suggest
+	if strings.HasPrefix(currentLineUpper, "FORMAT ") {
+		s = append(s, prompt.Suggest{Text: "table", Description: "Format Type"})
+		s = append(s, prompt.Suggest{Text: "json", Description: "Format Type"})
+		s = append(s, prompt.Suggest{Text: "csv", Description: "Format Type"})
+		return prompt.FilterFuzzy(s, d.GetWordBeforeCursor(), true)
+	} else if strings.HasPrefix(currentLineUpper, "USE ") {
+		for _, db := range c.Databases {
+			s = append(s, prompt.Suggest{Text: "\"" + db + "\"", Description: "Table Name"})
+		}
+		return prompt.FilterFuzzy(s, d.GetWordBeforeCursor(), true)
+	} else if strings.HasPrefix(currentLineUpper, "SELECT ") {
+		if isMatch, _ := regexp.Match(`FROM `+identRegex("from_clause")+`?$`, []byte(currentLineUpper)); isMatch {
+			if c.Database != "" && c.RetentionPolicy != "" {
+				for _, m := range c.Measurements {
+					s = append(s, prompt.Suggest{Text: "\"" + m + "\"", Description: fmt.Sprintf("Measurement on \"%s\".\"%s\"", c.Database, c.RetentionPolicy)})
+				}
+			}
+			if c.Database != "" {
+				for _, rp := range c.RetentionPolicies {
+					s = append(s, prompt.Suggest{Text: "\"" + rp + "\"", Description: "Retention Policy on " + c.Database})
+				}
+			}
+			for _, db := range c.Databases {
+				s = append(s, prompt.Suggest{Text: "\"" + db + "\"", Description: "Table Name"})
+			}
+			return prompt.FilterFuzzy(s, d.GetWordBeforeCursorUntilAnySeparator(" ", "."), true)
+		}
+	}
+	return append(
+		prompt.FilterHasPrefix(replKeywords, strings.ToLower(d.CurrentLine()), false),
+		prompt.FilterHasPrefix(allInfluxQLKeywords, strings.ToUpper(d.GetWordBeforeCursor()), true)...,
+	)
+}
+
+func (c *Client) help() {
+	fmt.Println(`Usage:
+        ` +
+		// ! UNSUPPORTED connect <host:port>   connects to another node specified by host:port
+		// ! UNSUPPORTED auth                  prompts for username and password
+		`pretty                toggles pretty print for the json format
+        use <db_name>         sets current database
```

Comment:
What about chunked responses?

Reply:
I wasn't aware of chunked responses available in the 2.x API, but I meant to leave a TODO in there to look into it further.

Reply:
I really hope we support chunked responses in 2.x - it's important to be able to return a response that doesn't fit it memory (both on the server and client side) by streaming the information one chunk at a time.

Reply:
It can be a TODO in this PR though.